### PR TITLE
fix: make zapper akschually great again

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Loosens Zapper schema some, removing some bits we're not concerned about, and allowing unknown keys.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes N/A

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Zapper read-only opportunities are great again

### Engineering

- Zapper spew should be gone in the app

## Screenshots (if applicable)

Develop: Missing zapper read-only opportunities and schema parsing spew

Screenshot 2024-08-01 at 21 51 05
image

This diff: happy Zapper opportunities 🎉 

Screenshot 2024-08-01 at 21 52 56
image

### CODE REVIEW

File: src/state/apis/zapper/validators.ts [hunk 1]
```diff
@@ -47,31 +47,6 @@ export enum SupportedZapperNetwork {
   Base = 'base',
 }
 
-enum SupportedZapperNetworkIncludeUnsupported {
```
Comment (gomesalexandre) [inline src/state/apis/zapper/validators.ts:50]: We actually don't need to know about all of these, only the supported ones

File: src/state/apis/zapper/validators.ts [hunk 2]
```diff
@@ -482,24 +455,24 @@ const V2AppGroupResponse = z.object({
...
+    farcasterUrl: z.string().nullable(),
+    tags: z.array(z.string()),
+    token: V2AppTokenResponse.nullable(),
+    groups: z.array(V2AppGroupResponse),
+  })
+  .allowUnknownKeys()
 
 export const V2AppsResponse = z.array(V2AppResponse)
 export type V2AppResponseType = Infer<typeof V2AppResponse>
-export type V2AppsResponseType = Infer<typeof V2AppsResponse>
```
Comment (gomesalexandre) [inline src/state/apis/zapper/validators.ts:505]: Unused

File: src/state/apis/zapper/validators.ts [hunk 3]
```diff
@@ -467,12 +445,7 @@ export type V2AppsBalancesResponseType = Infer<typeof V2AppsBalancesResponse>
 
 const V2AppTokenResponse = z.object({
   address: z.string(),
-  network: z.enum(SupportedZapperNetworkIncludeUnsupported),
-})
-
-const V2AppSupportedNetworkResponse = z.object({
-  network: z.enum(SupportedZapperNetworkIncludeUnsupported),
-  actions: z.array(z.string()),
+  network: z.string(),
```
Comment (gomesalexandre) [inline src/state/apis/zapper/validators.ts:448]: This'll do and will avoid us having to update the schema everytime zapper supports new networks - we have `ZAPPER_NETWORKS_TO_CHAIN_ID_MAP` and `CHAIN_ID_TO_ZAPPER_NETWORK_MAP` mappings either way

File: src/state/apis/zapper/validators.ts [hunk 4]
```diff
@@ -422,14 +400,14 @@ const userNftTokenSchema = z.object({
   items: z.array(userNftItemSchema).optional(),
 })
 
-const categorySchema = z.object({
-  id: z.number(),
-  name: z.string(),
-  slug: z.string(),
-  description: z.string(),
-  createdAt: z.string(),
```
Comment (gomesalexandre) [inline src/state/apis/zapper/validators.ts:430]: `createdAt` was an actual offender across the schema

File: src/state/apis/zapper/validators.ts [hunk 5]
```diff
@@ -422,14 +400,14 @@ const userNftTokenSchema = z.object({
...
-  updatedAt: z.string(),
```
Comment (gomesalexandre) [inline src/state/apis/zapper/validators.ts:431]: Not an offender, but unused

File: src/state/apis/zapper/validators.ts [hunk 6]
```diff
@@ -482,24 +455,24 @@ const V2AppGroupResponse = z.object({
...
-  slug: z.string(),
-  name: z.string(),
-  description: z.string(),
-  url: z.string(),
-  imgUrl: z.string(),
-  twitterUrl: z.string().nullable(),
-  farcasterUrl: z.string().nullable(),
-  tags: z.array(z.string()),
-  token: V2AppTokenResponse.nullable(),
-  supportedNetworks: z.array(V2AppSupportedNetworkResponse),
```
Comment (gomesalexandre) [inline src/state/apis/zapper/validators.ts:499]: We were actually not consuming this guy

**Review by kaladinlight — APPROVED** [review]
It's beautiful!

**Pull request merged**